ddtrace/internal/_encoding.pyx

```diff
@@ -273,37 +271,57 @@ cdef class Packer(object):
 
         return ret
 
-    cdef int _pack(self, object o) except -1:
+    cdef _pack_trace(self, list trace):
         cdef int ret
         cdef Py_ssize_t L
 
-        if o is None:
+        if trace is None:
```

Comment:
this should never be `None` right? can we assume that `list trace` is always a valid PyList? (it might be empty, but as long as it is known to be a PyList, then we should be good)